flink-libraries/flink-table/src/main/scala/org/apache/flink/table/catalog/ExternalCatalog.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog
+
+import java.util
+
+/**
+  * This class is responsible for interact with external catalog.
+  * Its main responsibilities including:
+  * <ul>
+  * <li> create/drop/alter database or tables for DDL operations
+  * <li> provide tables for calcite catalog, it looks up databases or tables in the external catalog
+  * </ul>
+  */
+trait ExternalCatalog {
+
+  /**
+    * Adds table into external Catalog
+    *
+    * @param table          description of table which to create
+    * @param ignoreIfExists whether to ignore operation if table already exists
+    */
+  def createTable(table: ExternalCatalogTable, ignoreIfExists: Boolean): Unit
```

Comment:
Hi, fabian. Even for the integration with Calcite, getTable(), getDataBase(), and listTables() are sufficient, however an complete ExternalCatalog should be responsible for CRUD operations on db/table. For some readonly ExternalCatalog, we could choose to not supported createX, dropX, andalterX methods , for example, throw UnsupportedOperationException.